src/workload.rs

```diff
@@ -900,64 +846,9 @@ pub struct WorkloadStore {
 }
 
 impl WorkloadStore {
-    #[cfg(test)]
-    pub fn test_store(workloads: Vec<XdsWorkload>) -> anyhow::Result<WorkloadStore> {
-        let mut store = WorkloadStore::default();
-        for w in workloads {
-            store.insert_xds_workload(w)?;
-        }
-        Ok(store)
-    }
-
-    pub fn full_test_store(
-        workloads: Vec<XdsWorkload>,
-        services: Vec<XdsService>,
-        authorizations: Vec<XdsAuthorization>,
-    ) -> anyhow::Result<WorkloadStore> {
-        let mut store = WorkloadStore::default();
-        for w in workloads {
-            store.insert_xds_workload(w)?;
-        }
-        for s in services {
-            store.insert_xds_service(s)?;
-        }
-        for a in authorizations {
-            store.insert_xds_authorization(a)?;
-        }
-        Ok(store)
-    }
-
-    fn insert_xds_address(&mut self, a: XdsAddress) -> anyhow::Result<()> {
-        match a.r#type {
-            Some(XdsType::Workload(w)) => self.insert_xds_workload(w),
-            Some(XdsType::Service(s)) => self.insert_xds_service(s),
-            _ => Err(anyhow::anyhow!("unknown address type")),
-        }
-    }
-
-    fn insert_xds_service(&mut self, s: XdsService) -> anyhow::Result<()> {
-        let svc: Service = Service::try_from(&s)?;
-        self.services.insert(svc);
-        Ok(())
-    }
-
-    fn insert_xds_workload(&mut self, w: XdsWorkload) -> anyhow::Result<()> {
-        let workload = Workload::try_from(&w)?;
-        self.insert_workload(workload.clone(), w.virtual_ips)?;
-
-        if self.should_prefetch_certificate(&workload) {
-            if let Some(tx) = self.cert_tx.as_mut() {
-                if let Err(e) = tx.try_send(workload.identity()) {
-                    info!("couldn't prefetch: {:?}", e)
-                }
-            }
-        }
-        Ok(())
-    }
-
     // Determine if we should prefetch a certificate for this workload. Being "wrong" is not too bad;
     // a missing cert will be fetched on-demand when we get a request, so will just result in some extra latency.
-    fn should_prefetch_certificate(&self, w: &Workload) -> bool {
+    fn should_prefetch_certificate(&self, w: Arc<Workload>) -> bool {
```

Comment:
not a huge fan of making this accept an Arc, i'd prefer to see the caller handle the lock

Reply:
This is only used within the struct, it's not a public API. The fact that it's an Arc means we don't need to copy.

Reply:
The latest code gets rid of the Arc.